src/state/queries/resolve-identity.ts

```diff
@@ -0,0 +1,27 @@
+import {LRU} from './direct-fetch-record'
+
+// TODO: lift this into direct fetch to share cache
+const serviceCache = new LRU<`did:${string}`, string>()
+
+export async function resolvePdsUrl(did: `did:${string}`) {
```

Comment:
can we rename this to `resolvePdsServiceUrl` for clarity? Big nitpick sorry, but I think it's more clear. There could be other types of urls in the did identity document, and in the future we might have a more general `didDocumentCache` to avoid refetching for those different queries...